### DISCUSSION TITLE
"Error: ENOENT: no such file or directory" with nested subfolder

### DISCUSSION THREAD

**Discussion by Lili1882** [nftchef/art-engine#327] [Help + Q&A]
Hello,
I need to use the structure with nested subfolder like te example, but when I execute the code, I have the following error:
 "Error: ENOENT: no such file or directory" with nested subfolder

I created Two Layers: "Background" and "Types" and dans Types the other subfolder "Alien, Human and Robot" etc... but when I execute the code I have un error , like there are not visibility for the other subfolder like Alien or Eyes or Skin..

<img width="1930" alt="Branching with nested folders (1)" src="https://github.com/nftchef/art-engine/assets/141786369/e5684691-c99b-4561-9615-839d7941508a">

Here the structure:

Layers:

Background
Types
    Alien#50
         ......Skin (Subfolder with attributes..)
         ......Eyes (Subfolder with attributes..)
   Human#30
  Robot#20

Thank you for your help!

**Comment by neilryanh** [comment]
Have you also adjusted the `src/config.js` to accommodate your folder/names?

**Reply by Lili1882** [reply]
Hello,
I found now the file Readme with the explication, so I did the modifications..

It's correct to put just the name of the sublayer without options, like below?

     {name: "Types",
        sublayerOptions: {name: "Alien#40"},},
       {name: "Alien",
        sublayerOptions: { name:"EYES"},
      },
      {name: "EYES",
      sublayerOptions: { name:"EYES#90"},
    },
................

Thank you!

**Reply by neilryanh** [reply]
You only have to add the initial folders to the `config.js`. In the example below:

> -layers
> ↳ Background
> ↳ Types
> -- ↳Alien#40
> -- ↳Human#30
> -- ↳Robot#20

Would look like: 
```
{
    growEditionSizeTo: 10,
    namePrefix: "Alien Human Robots", 
    layersOrder: [
      { name: "Background" },
      { name: "Types" },
    ],
  },
```
The rest will take care of itself. It'll run through and see the "#" weight on each sub-folder and/or file to process.

**Comment by Lili1882** [comment]
Hello,

And is it valable also for the subfolder EYES ? It’s like a second start
point and it has not the weight .. Is it included automatically or need to
add in the list of layers dans config.js?

Thank you for your answer.:)


Il giorno gio 10 ago 2023 alle 16:46 digital CREATOR <
***@***.***> ha scritto:

**Reply by neilryanh** [reply]
It should just work (it would be included). The first folders in `layers` are the start points and only those need to be entered in the `config.js`

Anything without a weight just means the generator is "forced" to go, or use.
